### PR TITLE
.Net: Fix issue #4608

### PR DESCRIPTION
**Pull request by Krzysztof318**
### Motivation and Context

Issue #4608

### Description

In multiple instances in the OpenAIMemoryBuilderExtensions class, the httpClient parameter was unintentionally hidden. To resolve this, discard was used in the function arguments to disable hiding. This change ensures the intended functionality isn't accidentally obstructed and enhances code readability and maintainability.

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

File: dotnet/src/Connectors/Connectors.OpenAI/OpenAIMemoryBuilderExtensions.cs [hunk 1]
```diff
@@ -33,7 +33,7 @@ public static class OpenAIMemoryBuilderExtensions
         string? modelId = null,
         HttpClient? httpClient = null)
     {
-        return builder.WithTextEmbeddingGeneration((loggerFactory, httpClient) =>
+        return builder.WithTextEmbeddingGeneration((loggerFactory, _) =>
```
Comment (SergeyMenshykh) [inline dotnet/src/Connectors/Connectors.OpenAI/OpenAIMemoryBuilderExtensions.cs:36]: This solution ignores the HTTP client set up by the `MemoryBuilder.WithHttpClient` method. Consider changing the parameter name for the builder HTTP client to `builderHttpClient` and use it first. If it's not provided, use the one from the `WithAzureOpenAITextEmbeddingGeneration` method. For example:
```C#
public static MemoryBuilder WithAzureOpenAITextEmbeddingGeneration(
    this MemoryBuilder builder,
    string deploymentName,
    string endpoint,
    string apiKey,
    string? modelId = null,
    HttpClient? httpClient = null)
{
    return builder.WithTextEmbeddingGeneration((loggerFactory, builderHttpClient) =>
        new AzureOpenAITextEmbeddingGenerationService(
            deploymentName,
            endpoint,
            apiKey,
            modelId,
            HttpClientProvider.GetHttpClient(builderHttpClient ?? httpClient),
            loggerFactory));
}
```

Comment (Krzysztof318) [reply dotnet/src/Connectors/Connectors.OpenAI/OpenAIMemoryBuilderExtensions.cs:36]: Ok, I can change this. But it's not better to give passed httpClient priority over builderHttpClient?

Comment (SergeyMenshykh) [reply dotnet/src/Connectors/Connectors.OpenAI/OpenAIMemoryBuilderExtensions.cs:36]: Correct, the HTTP client passed to the `WithAzureOpenAITextEmbeddingGeneration` should take precedence over the builder's one - `HttpClientProvider.GetHttpClient(httpClient ?? builderHttpClient)`.

Comment (Krzysztof318) [reply dotnet/src/Connectors/Connectors.OpenAI/OpenAIMemoryBuilderExtensions.cs]: Fixed :)

**Review by SergeyMenshykh — APPROVED** [review]

**Review by markwallace-microsoft — APPROVED** [review]

**Pull request merged**